### ISSUE TITLE
Add Shared classes/nested modules

### ISSUE THREAD

**Issue by jrmoreno1** [dotnet/vblang#440]
I like extension methods, but there’s a bit of a problem with them — quite frequently, they are only used inside a single class.  I would like to be able to declare a module/shared class inside another class and then make its methods extension methods.  

Unfortunately, you can’t declare a shared class at all, and modules can not be inside another class.  

I think VB ought to eliminate both of those limitations.

**Comment by JustNrik** [comment]
Having a `Shared Class` makes no sense when `Module` already exists, which is essentially a shared class with implicitly shared members. About nested modules, they don't make sense either, at least in c# which you can actually have nested static classes, having them is usually tagged code smell because they don't really make any sense.

**Comment by pricerc** [comment]
I'm sure there's been a request somewhere to allow extension methods to be declared in classes, which I think should be simpler to manage. 

I'd be happy if <Extension()> was valid on any shared method, even if only private ones (i.e. resolved purely by the compiler at compile time).

**Comment by jrmoreno1** [comment]
@pricerc : that is a much better solution.  I’ll look and if I find it, close this as a duplicate.

**Comment by paul1956** [comment]
I agree, I have loads of Extensions that are used in one Class and I end up having to create a Module move the extension there and the module serves no other purpose. And yes I would strongly prefer them private.

**Comment by bandleader** [comment]
Note also that in #165 I proposed to allow the use of any method as an extension method using `value.(methodName)`:

```vb
Shared Function Encode(value As String) As String
    ...
End Function

Sub DoWork(string As String)
    Dim encodedString = string.(Encode)
    ...
End Sub
```
